airgun/entities/oscapreports.py

```diff
@@ -0,0 +1,84 @@
+from time import sleep
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep
+from airgun.navigation import navigator
+from airgun.views.oscapreports import DeletereportsDialog
+from airgun.views.oscapreports import OSCAPReportsView
+
+
+class OSCAPReportsEntity(BaseEntity):
+    endpoint_path = '/compliance/arf_reports'
+
+    def search(self, value):
+        """Search for existing Compliance Reports."""
+        view = self.navigate_to(self, 'All')
+        return view.search(value)
+
+    def read_all(self):
+        """Read all values from Compliance Reports page."""
+        view = self.navigate_to(self, 'All')
+        return view.read()
+
+    def delete(self, search_pattern, cancel=False):
+        """Delete first Compliance Report based on search_pattern."""
+        view = self.navigate_to(self, 'All')
+        view.search(search_pattern)
+        view.table.row()['Actions'].widget.fill('Delete')
+        alert_message = self.browser.get_alert().text
+        self.browser.handle_alert(cancel=cancel)
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        return alert_message
+
+    def full_report(self, search_pattern, cancel=False):
+        """View Full Compliance Report."""
+        view = self.navigate_to(self, 'All')
+        view.search(search_pattern)
+        view.table.row()['Actions'].widget.fill('Full Report')
+        alert_message = self.browser.get_alert().text
+        self.browser.handle_alert(cancel=cancel)
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        return alert_message
+
+    def delete_reports(self, search_pattern):
+        """Delete all reports based on search_pattern
+           If keyword 'All' is supplied, all reports are selected
+           using the checkbox from table header
+        """
+        view = self.navigate_to(self, 'Delete reports', search_pattern=search_pattern)
+        sleep(1)
```

Comment:
Is this absolutely necessary? Can't we use something more explicit instead (what exactly we are waiting for)?

Also, that wait probably belongs to navigation endpoint `post_navigate` method. There are couple of usages in Airgun if you want to see how it could be implemented.

Reply:
No, this is not necessary. removed it.